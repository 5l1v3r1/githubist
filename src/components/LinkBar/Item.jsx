// @flow

import React, { type Node } from 'react';
import { NavLink, type LocationShape, type Match, type Location } from 'react-router-dom';
import cx from 'classnames';

import styles from './LinkBar.scss';

type Props = {
  children: Node,
  to: string | LocationShape,
  isActive?: (match: Match, location: Location) => boolean,
  exact?: boolean,
  outside?: boolean,
  className?: string,
  activeClassName?: string,
};

const Item = ({ children, to, isActive, exact, outside, className, activeClassName }: Props) => {
  const classNames = cx(styles.item, className);
  const activeClassNames = cx(styles.isActive, activeClassName);

  if (outside) {
    return (
      <a href={to} className={classNames} target="_blank" rel="noopener noreferrer">
        {children}
      </a>
    );
  }

  return (
    <NavLink
      to={to}
      className={classNames}
      isActive={isActive}
      exact={exact}
      activeClassName={activeClassNames}
    >
      {children}
    </NavLink>
  );
};

Item.defaultProps = {
  isActive: undefined,
  exact: false,
  outside: false,
  className: '',
  activeClassName: '',
};

export default Item;
