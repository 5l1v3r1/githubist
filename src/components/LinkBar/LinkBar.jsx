// @flow

import React, { type Node } from 'react';
import cx from 'classnames';

import styles from './LinkBar.scss';

type Props = {
  children: Node,
  withoutGap?: boolean,
  className?: string,
};

const LinkBar = ({ children, withoutGap, className }: Props) => {
  const classNames = cx(styles.linkBar, { [styles.withoutGap]: withoutGap }, className);

  return <div className={classNames}>{children}</div>;
};

LinkBar.defaultProps = {
  withoutGap: false,
  className: '',
};

export default LinkBar;
