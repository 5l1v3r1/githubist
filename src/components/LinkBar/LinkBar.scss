@import '../../theme/theme';

.linkBar {
  display: flex;
  flex-direction: row;
  padding-bottom: map-get($spacers, x-large);
}

.withoutGap {
  padding-bottom: 0;
}

.item {
  margin-right: map-get($spacers, normal);
  color: $text-muted;
  font-size: map-get($font-sizes, large);
  text-decoration: none;

  &:hover,
  &.isActive {
    color: $black-color;
  }
}
