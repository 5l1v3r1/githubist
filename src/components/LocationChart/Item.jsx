// @flow

import React, { type Node } from 'react';

import styles from './LocationChart.scss';

type Props = {
  rank: number,
  percentage: number,
  children: Node,
};

const Item = ({ children, rank, percentage }: Props) => (
  <div className={styles.item}>
    <div className={styles.rank}>{rank}</div>
    <div className={styles.outer}>
      <div className={styles.bar} style={{ width: `${percentage.toString()}%` }}>
        <div className={styles.content}>{children}</div>
      </div>
    </div>
  </div>
);

export default Item;
