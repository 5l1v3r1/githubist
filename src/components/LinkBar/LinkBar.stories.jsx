// @flow

import React from 'react';
import { storiesOf } from '@storybook/react';
import backgrounds from '@storybook/addon-backgrounds';
import { MemoryRouter } from 'react-router-dom';

import LinkBar from './LinkBar';
import Item from './Item';

storiesOf('LinkBar', module)
  .addDecorator(backgrounds([{ name: 'dark', value: '#2d2e33', default: true }]))
  .add('default', () => (
    <div style={{ backgroundColor: '#fff' }}>
      <MemoryRouter>
        <LinkBar>
          <Item to="/link-1" isActive={() => true}>
            Link 1
          </Item>
          <Item to="/link-2">Link 2</Item>
          <Item to="/link-3">Link 3</Item>
        </LinkBar>
      </MemoryRouter>
    </div>
  ))
  .add('with outside link', () => (
    <div style={{ backgroundColor: '#fff' }}>
      <MemoryRouter>
        <LinkBar>
          <Item to="/link-1" isActive={() => true}>
            Link 1
          </Item>
          <Item to="/link-2">Link 2</Item>
          <Item to="/link-3">Link 3</Item>
          <Item to="https://github.com" outside>
            Github
          </Item>
        </LinkBar>
      </MemoryRouter>
    </div>
  ))
  .add('without gap', () => (
    <div style={{ backgroundColor: '#fff' }}>
      <MemoryRouter>
        <LinkBar withoutGap>
          <Item to="/link-1" isActive={() => true}>
            Link 1
          </Item>
          <Item to="/link-2">Link 2</Item>
          <Item to="/link-3">Link 3</Item>
        </LinkBar>
      </MemoryRouter>
    </div>
  ));
