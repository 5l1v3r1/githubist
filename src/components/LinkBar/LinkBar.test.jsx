// @flow

import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { mount } from 'enzyme';

import LinkBar from './LinkBar';
import Item from './Item';

it('renders correctly', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkBar>
        <Item to="/link-1">Link 1</Item>
        <Item to="/link-2">Link 2</Item>
        <Item to="/link-3">Link 3</Item>
      </LinkBar>
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('renders correctly with active item', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkBar>
        <Item to="/link-1" isActive={() => true}>
          Link 1
        </Item>
        <Item to="/link-2">Link 2</Item>
        <Item to="/link-3">Link 3</Item>
      </LinkBar>
    </MemoryRouter>,
  );

  expect(wrapper).toMatchSnapshot();
});

it('passes custom classnames to container', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkBar className="testClassName">
        <Item to="/link-1">Link 1</Item>
      </LinkBar>
    </MemoryRouter>,
  );

  expect(wrapper.find('div').hasClass('testClassName')).toBe(true);
});

it('passes custom classnames to item', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkBar>
        <Item to="/link-1" className="testClassName">
          Link 1
        </Item>
      </LinkBar>
    </MemoryRouter>,
  );

  expect(wrapper.find('a').hasClass('testClassName')).toBe(true);
});

it('passes custom active classnames to item', () => {
  const wrapper = mount(
    <MemoryRouter keyLength={0}>
      <LinkBar>
        <Item to="/link-1" activeClassName="testClassName" isActive={() => true}>
          Link 1
        </Item>
      </LinkBar>
    </MemoryRouter>,
  );

  expect(wrapper.find('a').hasClass('testClassName')).toBe(true);
});
