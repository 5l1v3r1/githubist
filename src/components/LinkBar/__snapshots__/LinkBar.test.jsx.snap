// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkBar
      className=""
      withoutGap={false}
    >
      <div
        className=""
      >
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-1"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-1"
          >
            <Route
              exact={false}
              path="/link-1"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-1"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-1"
                  onClick={[Function]}
                >
                  Link 1
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-2"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-2"
          >
            <Route
              exact={false}
              path="/link-2"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-2"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-2"
                  onClick={[Function]}
                >
                  Link 2
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-3"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-3"
          >
            <Route
              exact={false}
              path="/link-3"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-3"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-3"
                  onClick={[Function]}
                >
                  Link 3
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
      </div>
    </LinkBar>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly with active item 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkBar
      className=""
      withoutGap={false}
    >
      <div
        className=""
      >
        <Item
          activeClassName=""
          className=""
          exact={false}
          isActive={[Function]}
          to="/link-1"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            isActive={[Function]}
            to="/link-1"
          >
            <Route
              exact={false}
              path="/link-1"
            >
              <Link
                aria-current="true"
                className=""
                replace={false}
                style={Object {}}
                to="/link-1"
              >
                <a
                  aria-current="true"
                  className=""
                  href="/link-1"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Link 1
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-2"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-2"
          >
            <Route
              exact={false}
              path="/link-2"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-2"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-2"
                  onClick={[Function]}
                >
                  Link 2
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-3"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-3"
          >
            <Route
              exact={false}
              path="/link-3"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-3"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-3"
                  onClick={[Function]}
                >
                  Link 3
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
      </div>
    </LinkBar>
  </Router>
</MemoryRouter>
`;

exports[`renders correctly without bottom gap 1`] = `
<MemoryRouter
  keyLength={0}
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <LinkBar
      className=""
      withoutGap={true}
    >
      <div
        className="undefined"
      >
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-1"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-1"
          >
            <Route
              exact={false}
              path="/link-1"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-1"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-1"
                  onClick={[Function]}
                >
                  Link 1
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-2"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-2"
          >
            <Route
              exact={false}
              path="/link-2"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-2"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-2"
                  onClick={[Function]}
                >
                  Link 2
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
        <Item
          activeClassName=""
          className=""
          exact={false}
          to="/link-3"
        >
          <NavLink
            activeClassName=""
            ariaCurrent="true"
            className=""
            exact={false}
            to="/link-3"
          >
            <Route
              exact={false}
              path="/link-3"
            >
              <Link
                aria-current={false}
                className=""
                replace={false}
                to="/link-3"
              >
                <a
                  aria-current={false}
                  className=""
                  href="/link-3"
                  onClick={[Function]}
                >
                  Link 3
                </a>
              </Link>
            </Route>
          </NavLink>
        </Item>
      </div>
    </LinkBar>
  </Router>
</MemoryRouter>
`;
